src/gradualizer_type.erl

```diff
@@ -364,4 +365,8 @@
 
 -type type_name() :: atom().
 
+-type af_constraint() :: {'type', anno(), 'constraint',
```

Comment:
For everything that is not copied from OTP here,  please add a comment about it.

Don't export the af types but add an abstract_constraint typedef as for the other exported ones.

Reply:
This is copied from OTP, it's a part of `erl_parse:abstract_form()`. To my surprise, the `abstract_form` is the only `abstract_*` thing not copied from `erl_parse` to `gradualizer_type` verbatim 🤷

> Don't export the af types but add an abstract_constraint typedef as for the other exported ones.

I could do this, but then we're deviating from the naming used in `erl_parse`. Do you think that's ok?

Reply:
> (...) surprise (...) copied (...) verbatim

We needed to copy the whole module to be able to export types not exported by default.

> Do you think that's ok?

I changed my mind. You can export the `af_` name. I see now that `erl_parse` has a few exported `af_` types: https://www.erlang.org/doc/man/erl_parse.html#data-types